From 40b5bbb91e1077ff0b7d88d7cbcff0228e282616 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Thu, 3 Jan 2013 12:57:58 -0800
Subject: [PATCH 1/6] add Belarussian mapping: simple, 'by' and 'by'

---
 src/locale/kbd-model-map | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/locale/kbd-model-map b/src/locale/kbd-model-map
index a895880..a0d7d19 100644
--- a/src/locale/kbd-model-map
+++ b/src/locale/kbd-model-map
@@ -70,3 +70,4 @@ cf			ca(fr)	pc105		-		terminate:ctrl_alt_bksp
 sv-latin1		se	pc105		-		terminate:ctrl_alt_bksp
 sr-cy			rs	pc105		-		terminate:ctrl_alt_bksp
 gr			gr,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
+by			by,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
-- 
1.8.0.2

