From 07f8a4aa49a84ec61513788d5ddf521f3de5a0ba Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 1 Aug 2011 20:51:18 +0200
Subject: [PATCH] manager: call generators with umask 0022

---
 src/manager.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/manager.c b/src/manager.c
index 9e4bd51..69dbec4 100644
--- a/src/manager.c
+++ b/src/manager.c
@@ -2985,6 +2985,7 @@ void manager_run_generators(Manager *m) {
         DIR *d = NULL;
         const char *generator_path;
         const char *argv[3];
+        mode_t u;
 
         assert(m);
 
@@ -3027,7 +3028,9 @@ void manager_run_generators(Manager *m) {
         argv[1] = m->generator_unit_path;
         argv[2] = NULL;
 
+        u = umask(0022);
         execute_directory(generator_path, d, (char**) argv);
+        umask(u);
 
         if (rmdir(m->generator_unit_path) >= 0) {
                 /* Uh? we were able to remove this dir? I guess that
-- 
1.7.6

