From 2eaaf10fe053f1e2f1668319d5cd13328500be7d Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Thu, 3 Jan 2013 13:20:08 -0800
Subject: [PATCH 3/6] add Hebrew (Israel) - simple, il/il

---
 src/locale/kbd-model-map | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/locale/kbd-model-map b/src/locale/kbd-model-map
index e05322d..ee8f9c1 100644
--- a/src/locale/kbd-model-map
+++ b/src/locale/kbd-model-map
@@ -71,3 +71,4 @@ sv-latin1		se	pc105		-		terminate:ctrl_alt_bksp
 sr-cy			rs	pc105		-		terminate:ctrl_alt_bksp
 gr			gr,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
 by			by,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
+il			il	pc105		-		terminate:ctrl_alt_bksp
-- 
1.8.0.2

