From 926f734836ab0b67e62d55b54e503728da30cee1 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 11 Jan 2013 23:47:11 +0100
Subject: [PATCH] (F18) bring back single.service

---
 Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index ce56d05..18b9c4c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4011,7 +4011,8 @@ endif
 SYSTEM_UNIT_ALIASES += \
 	graphical.target default.target \
 	reboot.target ctrl-alt-del.target \
-	getty@.service autovt@.service
+	getty@.service autovt@.service \
+	rescue.service single.service
 
 USER_UNIT_ALIASES += \
 	$(systemunitdir)/shutdown.target shutdown.target \
