From 644f61739799fa3c4b9a0e7d56b48bc386705138 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 11 Jan 2013 23:47:11 +0100
Subject: [PATCH] (F18) bring back single.service

---
 Makefile.am | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 26d4672..28d8d44 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4080,6 +4080,10 @@ if HAVE_KMOD
 		$(LN_S) ../systemd-modules-load.service systemd-modules-load.service )
 endif
 
+	( cd $(DESTDIR)$(systemunitdir) && \
+		rm -f single.service && \
+		$(LN_S) rescue.service single.service )
+
 install-exec-hook: $(INSTALL_EXEC_HOOKS)
 
 uninstall-hook: $(UNINSTALL_DATA_HOOKS) $(UNINSTALL_EXEC_HOOKS)
