From 12939e721759aeeb24830ac0904ed8a701c07e24 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Thu, 3 Jan 2013 13:28:41 -0800
Subject: [PATCH 5/6] add Lithuanian keyboard mapping (lt/lt)

---
 src/locale/kbd-model-map | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/locale/kbd-model-map b/src/locale/kbd-model-map
index 500b688..1688373 100644
--- a/src/locale/kbd-model-map
+++ b/src/locale/kbd-model-map
@@ -73,3 +73,4 @@ gr			gr,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
 by			by,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
 il			il	pc105		-		terminate:ctrl_alt_bksp
 kazakh			kz,us	pc105		-		terminate:ctrl_alt_bksp,grp:shifts_toggle,grp_led:scroll
+lt			lt	pc105		-		terminate:ctrl_alt_bksp
-- 
1.8.0.2

