From b09b841398cccd3794b9573d57828e810f5f33cb Mon Sep 17 00:00:00 2001
From: WANG Chao <chaowang@redhat.com>
Date: Fri, 9 Aug 2013 17:01:50 +0800
Subject: [PATCH 14/25] fstab-generator: read rd.fstab=on/off switch correctly

---
 src/fstab-generator/fstab-generator.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index c17299f..479938f 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -596,9 +596,9 @@ static int parse_proc_cmdline(void) {
                 } else if (startswith(word, "rd.fstab=")) {
 
                         if (in_initrd()) {
-                                r = parse_boolean(word + 6);
+                                r = parse_boolean(word + 9);
                                 if (r < 0)
-                                        log_warning("Failed to parse fstab switch %s. Ignoring.", word + 6);
+                                        log_warning("Failed to parse fstab switch %s. Ignoring.", word + 9);
                                 else
                                         arg_enabled = r;
                         }
-- 
1.8.3.1

