From 9a11d4dc5b77fc3b9b33f96888f1a46f5a013786 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Thu, 3 Mar 2011 01:30:08 +0100
Subject: [PATCH] systemadm: display dependencies sorted (cherry picked from
 commit 8278f06953f5339646e5ff98900321f1525c0a21)

---
 src/systemadm.vala |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/systemadm.vala b/src/systemadm.vala
index c893da0..088ba26 100644
--- a/src/systemadm.vala
+++ b/src/systemadm.vala
@@ -458,6 +458,10 @@ public class MainWindow : Window {
         }
 
         public string make_dependency_string(string? prefix, string word, string[] dependencies) {
+                Gee.Collection<unowned string> sorted = new Gee.TreeSet<string>();
+                foreach (string i in dependencies)
+                        sorted.add(i);
+
                 bool first = true;
                 string r;
 
@@ -466,7 +470,7 @@ public class MainWindow : Window {
                 else
                         r = prefix;
 
-                foreach (string i in dependencies) {
+                foreach (string i in sorted) {
                         if (r != "")
                                 r += first ? "\n" : ",";
 
-- 
1.7.7.5

