From 0d380f2f2d27eabcab8167a52e2d7196720f3d6a Mon Sep 17 00:00:00 2001
From: Gregs Gregs <gregpuppy01@yahoo.gr>
Date: Mon, 14 Nov 2011 19:58:03 +0200
Subject: [PATCH] fsck: Fix typo in comment (cherry picked from commit
 b911442003350c56673e5689328f173ed03bbabd)

---
 src/fsck.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/fsck.c b/src/fsck.c
index c5088ad..3e66535 100644
--- a/src/fsck.c
+++ b/src/fsck.c
@@ -80,7 +80,7 @@ static void start_target(const char *target, bool isolate) {
 
         if (!(reply = dbus_connection_send_with_reply_and_block(bus, m, -1, &error))) {
 
-                /* Don't print a waring if we aren't called during
+                /* Don't print a warning if we aren't called during
                  * startup */
                 if (!dbus_error_has_name(&error, BUS_ERROR_NO_SUCH_JOB))
                         log_error("Failed to start unit: %s", bus_error_message(&error));
-- 
1.7.7.5

