From d03880f0c35228ca1b4d816b28c606a4e41d1698 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Thu, 19 Sep 2013 14:55:35 -0400
Subject: [PATCH] fix grammatical error

---
 src/nspawn/nspawn.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nspawn/nspawn.c b/src/nspawn/nspawn.c
index fbbe52c..839c253 100644
--- a/src/nspawn/nspawn.c
+++ b/src/nspawn/nspawn.c
@@ -410,7 +410,7 @@ static int mount_binds(const char *dest, char **l, unsigned long flags) {
 
                 if (stat(where, &dest_st) == 0) {
                         if ((source_st.st_mode & S_IFMT) != (dest_st.st_mode & S_IFMT)) {
-                                log_error("The file types of %s and %s do not matching. Refusing bind mount",
+                                log_error("The file types of %s and %s do not match. Refusing bind mount",
                                                 *x, where);
                                 return -EINVAL;
                         }
