From 271f93368eee2784ad360fe6cfed0b684f4a1032 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 19 Dec 2011 13:12:36 +0100
Subject: [PATCH] man: build new man pages (cherry picked from commit
 f0d2e205a28e37528ef791cc2913e6664d0dde7f)

---
 Makefile.am |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4d04db3..3b2cbc2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -818,6 +818,12 @@ MANPAGES_ALIAS = \
 	man/sd_is_socket_unix.3 \
 	man/sd_is_socket_inet.3 \
 	man/sd_notifyf.3 \
+        man/sd_pid_get_session.3 \
+        man/sd_uid_get_state.3 \
+        man/sd_session_is_active.3 \
+        man/sd_seat_get_active.3 \
+        man/sd_get_seats.3 \
+        man/sd_login_monitor_new.3 \
 	man/init.1
 
 man/reboot.8: man/halt.8
-- 
1.7.7.5

