From a00aab29d1970ef575ee395b016363cf48d39a7d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 3 Jan 2012 21:50:02 +0100
Subject: [PATCH] logind: don't watch vcsa if nobody cares (cherry picked from
 commit 976c088a0242a1a91b8f00899a4c2ae0e621a185)

---
 src/logind.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/logind.c b/src/logind.c
index 3a1903c..333d5f8 100644
--- a/src/logind.c
+++ b/src/logind.c
@@ -955,7 +955,8 @@ static int manager_connect_udev(Manager *m) {
         ev.events = EPOLLIN;
         ev.data.u32 = FD_SEAT_UDEV;
 
-        if (m->n_autovts <= 0)
+        /* Don't bother watching VCSA devices, if nobody cares */
+        if (m->n_autovts <= 0 || m->console_active_fd < 0)
                 return 0;
 
         if (epoll_ctl(m->epoll_fd, EPOLL_CTL_ADD, m->udev_seat_fd, &ev) < 0)
@@ -1074,13 +1075,13 @@ int manager_startup(Manager *m) {
         if (m->epoll_fd < 0)
                 return -errno;
 
-        /* Connect to udev */
-        r = manager_connect_udev(m);
+        /* Connect to console */
+        r = manager_connect_console(m);
         if (r < 0)
                 return r;
 
-        /* Connect to console */
-        r = manager_connect_console(m);
+        /* Connect to udev */
+        r = manager_connect_udev(m);
         if (r < 0)
                 return r;
 
-- 
1.7.7.5

