From 427275bed23af1079fc9140459fea61e5d06e7fc Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Thu, 5 Jan 2012 00:52:56 +0100
Subject: [PATCH] logind: add needed include for sd_notify() (cherry picked
 from commit 4726299357861c0ced3a53a4214e1c3809968e5e)

---
 src/logind.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/logind.c b/src/logind.c
index 333d5f8..5cff2db 100644
--- a/src/logind.c
+++ b/src/logind.c
@@ -30,6 +30,7 @@
 #include <linux/vt.h>
 
 #include "logind.h"
+#include "sd-daemon.h"
 #include "dbus-common.h"
 #include "dbus-loop.h"
 #include "strv.h"
