From a2aa71a9542492d6b34007b9dac28e4f10b4256d Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Mon, 14 Sep 2015 09:26:19 +0200
Subject: [PATCH] keymap: Add Corsair K70

---
 hwdb/60-keyboard.hwdb | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hwdb/60-keyboard.hwdb b/hwdb/60-keyboard.hwdb
index 3e49449..32d39dd 100644
--- a/hwdb/60-keyboard.hwdb
+++ b/hwdb/60-keyboard.hwdb
@@ -944,6 +944,14 @@ evdev:input:b0003v047FpC006*
  KEYBOARD_KEY_b002f=f20                                # Microphone mute button; should be micmute
 
 ###########################################################
+# Corsair
+###########################################################
+
+# Corsair K70
+evdev:input:b0003v1B1Cp1B09*
+ KEYBOARD_KEY_70032=reserved
+
+###########################################################
 # Quanta
 ###########################################################
 
