diff -up systemd-37/src/tty-ask-password-agent.c.744415 systemd-37/src/tty-ask-password-agent.c
--- systemd-37/src/tty-ask-password-agent.c.744415	2011-11-03 17:01:25.000000000 +0100
+++ systemd-37/src/tty-ask-password-agent.c	2011-11-03 17:02:18.000000000 +0100
@@ -206,6 +206,7 @@ static int ask_password_plymouth(
                                 continue;
 
                         memcpy(&size, buffer+1, sizeof(size));
+                        size = le32toh(size);
                         if (size+5 > sizeof(buffer)) {
                                 r = -EIO;
                                 goto finish;
