diff -up systemd-stable-246.9/src/nss-resolve/nss-resolve.c.fallback systemd-stable-246.9/src/nss-resolve/nss-resolve.c
--- systemd-stable-246.9/src/nss-resolve/nss-resolve.c.fallback	2021-01-04 15:48:33.668757361 -0500
+++ systemd-stable-246.9/src/nss-resolve/nss-resolve.c	2021-01-04 15:49:00.111891935 -0500
@@ -23,7 +23,8 @@ NSS_GETHOSTBYNAME_PROTOTYPES(resolve);
 NSS_GETHOSTBYADDR_PROTOTYPES(resolve);
 
 static bool bus_error_shall_fallback(sd_bus_error *e) {
-        return sd_bus_error_has_name(e, SD_BUS_ERROR_SERVICE_UNKNOWN) ||
+        return sd_bus_error_get_errno(e) == ENOTCONN ||
+               sd_bus_error_has_name(e, SD_BUS_ERROR_SERVICE_UNKNOWN) ||
                sd_bus_error_has_name(e, SD_BUS_ERROR_NAME_HAS_NO_OWNER) ||
                sd_bus_error_has_name(e, SD_BUS_ERROR_NO_REPLY) ||
                sd_bus_error_has_name(e, SD_BUS_ERROR_ACCESS_DENIED) ||
